src/librustc/mir/tcx.rs

```diff
@@ -149,7 +149,9 @@ impl<'tcx> Rvalue<'tcx> {
     {
         match *self {
             Rvalue::Use(ref operand) => operand.ty(local_decls, tcx),
-            Rvalue::Repeat(ref operand, count) => tcx.mk_array(operand.ty(local_decls, tcx), count),
+            Rvalue::Repeat(ref operand, count) => {
+                tcx.mk_ty(ty::Array(operand.ty(local_decls, tcx), count))
```

Comment:
I wonder when we can have `mk_array` take a `ty::Const`.

Reply:
Likely never, we have 14 uses in the code base, and half of them are `ProjectionKind::SubSlice` related.